src/PostgREST/Config.hs

```diff
@@ -331,22 +330,18 @@ instance JustIfMaybe a (Maybe a) where
   justIfMaybe a = Just a
 
 -- | Parse the config file
-readAppConfig :: [(Text, Text)] -> Environment -> Maybe FilePath -> IO AppConfig
+readAppConfig :: [(Text, Text)] -> Environment -> Maybe FilePath -> IO (Either Text AppConfig)
 readAppConfig dbSettings env optPath = do
   -- Now read the actual config file
   conf <- case optPath of
-    Just cfgPath -> catches (C.load cfgPath)
-      [ Handler (\(ex :: IOError)    -> exitErr $ "Cannot open config file:\n\t" <> show ex)
-      , Handler (\(C.ParseError err) -> exitErr $ "Error parsing config file:\n" <> err)
+    Just cfgPath -> C.load cfgPath `catches`
+      [ Handler (\(ex :: IOError)    -> panic $ "Cannot open config file: " <> show ex)
+      , Handler (\(C.ParseError err) -> panic $ "Error parsing config file: " <> err)
```

Comment:
Ideally these panics would go in Main, but I wasn't able to change them to Lefts.

They're only likely to occur on startup though(not during reload), so I thought it's fine to leave them for now.

Reply:
> They're only likely to occur on startup though(not during reload), so I thought it's fine to leave them for now.

Not sure I understand that reasoning. Whenever somebody deletes the config file and then tries to reload - or when they change the config file to something with invalid syntax and then reloads - those will occur. This seems equally likely on startup and reload.

Reply:
You're right. I've found that I can replace the `catches` with [try](https://hackage.haskell.org/package/base-4.14.1.0/docs/Control-Exception.html#v:try). I'll make a PR with that change.